### PR TITLE
pschnayder: Updating the StreamWriterClosedException to include client side mitigation suggestions from the public documentation

### PR DESCRIPTION
**Pull request by Philipendulum**
Improved error message allows end users to have actionable remediation for encountered com.google.api.gax.rpc.UnavailableException based on https://cloud.google.com/java/docs/reference/gax/latest/com.google.api.gax.rpc.UnavailableException

The feature request is created here -> https://github.com/googleapis/java-bigquerystorage/issues/2783

### CODE REVIEW

**Review by whuffman36 — APPROVED** [review]
LGTM, nice work!

File: google-cloud-bigquerystorage/src/main/java/com/google/cloud/bigquery/storage/v1/ConnectionWorker.java [hunk 1]
```diff
@@ -585,11 +585,21 @@ private ApiFuture<AppendRowsResponse> appendInternal(
       }
 
       if (connectionFinalStatus != null) {
+        String connectionFinalStatusString;
+        if (connectionFinalStatus
+            .toString()
+            .contains("com.google.api.gax.rpc.UnavailableException")) {
+          connectionFinalStatusString =
+              connectionFinalStatus.toString()
+                  + " This is a most likely a transient condition and may be corrected by retrying"
```
[the remark this answers is not in the document]
Comment (Philipendulum) [reply google-cloud-bigquerystorage/src/main/java/com/google/cloud/bigquery/storage/v1/ConnectionWorker.java:594]: It doesn't end with punctuation. I'll put a period before " This..."

**Comment by whuffman36** [comment]
kokoro:run

**Pull request merged**